### PR TITLE
drawBaseline for barplots and stemseries

### PR DESCRIPTION
**Pull request by okomarov**
Now draws baseline in barplots and stemseries plots:

ACID(32)
capture

### CODE REVIEW

File: src/matlab2tikz.m [hunk 1]
```diff
@@ -3710,6 +3710,9 @@ function warnFacetedInterp(m2t)
     str = sprintf('\\addplot[%s] plot table[%s] {%s};\n', ...
                  opts_print(m2t, drawOptions, ','), ...
                  opts_print(m2t, tabOpts, ','), table);
+
+    [m2t, baseline] = drawBaseline(m2t,h,isHoriz);
```
Comment (egeerardyn) [inline src/matlab2tikz.m]: What bothers me a bit is that you pass `isHoriz` to a parameter called `isVertical`. I guess `isHoriz` means the bars are in a horizontal direction, so the base line should be vertical.
I think that should become apparent from the code as well (if my explanation is correct, of course).

File: src/matlab2tikz.m [hunk 2]
```diff
@@ -3842,8 +3845,52 @@ function warnFacetedInterp(m2t)
     [m2t, drawOptions] = setColor(m2t, h, drawOptions, 'fill', faceColor);
 end
 % ==============================================================================
+function [m2t,str] = drawBaseline(m2t,hparent,isVertical)
+    str = '';
```
Comment (egeerardyn) [inline src/matlab2tikz.m]: Could do with some help text.

File: src/matlab2tikz.m [hunk 3]
```diff
@@ -3842,8 +3845,52 @@ function warnFacetedInterp(m2t)
...
+    baseValue = get(hparent, 'BaseValue');
+    if isOff(get(hparent,'ShowBaseLine')) || ~isHG2() || baseValue ~= 0
```
Comment (egeerardyn) [inline src/matlab2tikz.m]: Why don't we draw the line when `baseValue == 0`? That should be documented in the code (or we should just draw it).

File: src/matlab2tikz.m [hunk 4]
```diff
@@ -3842,8 +3845,52 @@ function warnFacetedInterp(m2t)
...
+        return
+    end
+    
+    if ~exist('isVertical','var')
```
Comment (egeerardyn) [inline src/matlab2tikz.m]: Please put this at the top of the function, that makes sure that `isVertical` is available in the whole function.

**Comment by egeerardyn** [comment]
Nice work already!

File: src/matlab2tikz.m [hunk 5]
```diff
@@ -3842,8 +3845,52 @@ function warnFacetedInterp(m2t)
...
+        xData = repmat(baseValue,1,2);
+        yData = get(m2t.currentHandles.gca,'Ylim');
+    else
+        xData = get(m2t.currentHandles.gca,'Xlim');
+        yData = repmat(baseValue,1,2);
+    end
+    
+    [m2t, table, tabOpts] = makeTable(m2t, '', xData, '', yData);
+
+    str = sprintf('%s\\addplot[%s] plot table[%s] {%s};\n', ...
```
Comment (egeerardyn) [inline src/matlab2tikz.m]: Shouldn't this be `%s\\addplot[%s] table[%s] {%s};\n` ? I know we have the `plot table` somewhere else in the code as well (and it compiles just fine), but in the `pgfplots` manual I can only find `\addplot table`, never `\addplot plot table`, though `plot table` is used as a search term in the manual.

What do you think?

File: src/matlab2tikz.m (hunk 1 above)
Comment (okomarov) [inline src/matlab2tikz.m]: Yes, you are correct. I was testing if you were paying attention :laughing:!
I will add comments, that I usually do at the end unless the code is hard to understand.

File: src/matlab2tikz.m (hunk 3 above)
Comment (okomarov) [inline src/matlab2tikz.m]: U mean when ~= 0?

File: src/matlab2tikz.m (hunk 4 above)
Comment (okomarov) [inline src/matlab2tikz.m]: :+1:

File: src/matlab2tikz.m (hunk 5 above)
Comment (okomarov) [inline src/matlab2tikz.m]: Will have to look into it, copy-pasted from stemplot if I recall correctly.

File: src/matlab2tikz.m (hunk 3 above)
Comment (egeerardyn) [inline src/matlab2tikz.m]: Oh, indeed. But the question remains the same (well, actually replace `==` with `~=`, but you get the point).

File: src/matlab2tikz.m (hunk 1 above)
Comment (egeerardyn) [inline src/matlab2tikz.m]: :+1:

**Comment by egeerardyn** [comment]
:+1: Feel free to merge when you're done with this.

**Comment by PeterPablo** [comment]
Code looks good (did not runtime test)! :+1: 
Is HG1 not affected, since the baseline is already dealt with implicitly?

**Comment by okomarov** [comment]
@PeterPablo Yes, since the baseline is a line object and the handleAllChildren() detects it.

**Pull request merged**